src/Test/L1/Worker/WorkerL1Tests.cs

```diff
@@ -34,6 +36,7 @@ public async Task Test_Base()
             {
                 Assert.Equal(expectedSteps[idx], steps[idx].Name);
             }
+            TearDown();
```

Comment:
If disposing the l1hostcontext is really important, TearDown should be in a finally. If the assert fails, teardown won't happen. This is true for all the test methods. You may want to create a RunTest method takes the "action" to test and does the setup and tear down.

Reply:
Added try/finally to stay consistent with other parts of the code.